### PR TITLE
fix(templates): add identity ref patch to kustomize

### PR DESCRIPTION
**Pull request by sonasingh46**
Signed-off-by: Ashutosh Kumar <sonasingh46@gmail.com>

 

 
**What type of PR is this?**
This PR adds identity ref patch to kustomize. 

/kind bug

**What this PR does / why we need it**:
This PRs fix a bug for system-assigned-identity flavor where the identity ref was not update in the AzureCluster spec. 

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Fix template for system-assigned-identiy flavor 
```

### CODE REVIEW

File: templates/flavors/system-assigned-identity/kustomization.yaml [hunk 1]
```diff
@@ -4,3 +4,5 @@ resources:
   - ../default/machine-deployment.yaml
 patchesStrategicMerge:
   - patches/system-assigned-identity.yaml
+  - ../../azure-cluster-identity/azurecluster-identity-ref.yaml
+
```
Comment (CecileRobertMichon) [inline templates/flavors/system-assigned-identity/kustomization.yaml:8]: nit: extra line

File: templates/cluster-template-system-assigned-identity.yaml [hunk 2]
```diff
@@ -25,6 +25,10 @@ metadata:
   name: ${CLUSTER_NAME}
   namespace: default
 spec:
+  identityRef:
```
Comment (CecileRobertMichon) [inline templates/cluster-template-system-assigned-identity.yaml:28]: while you're here, could you please also add it to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/flavors/user-assigned-identity/kustomization.yaml which has the same issue?

Comment (sonasingh46) [reply templates/cluster-template-system-assigned-identity.yaml:28]: done

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/approve

**Pull request merged**